tests/test_controller.py

```diff
@@ -2423,257 +2429,282 @@ def test_recommendations(self):
-        [entry1, entry2] = feed['entries']
-        eq_(self.work.title, entry1['title'])
-        eq_(another_work.title, entry2['title'])
-
-        metadata.recommendations = [
-            self.work.license_pools[0].identifier,
-            another_work.license_pools[0].identifier,
-        ]
-        mock_api.setup(metadata)
+        # The RecommendationLane is set up to ask for recommendations
+        # for this book.
+        lane = kwargs.pop('lane')
+        assert isinstance(lane, RecommendationLane)
+        library = self._default_library
+        eq_(library.id, lane.library_id)
+        eq_(self.english_1, lane.work)
+        eq_('Recommendations for Quite British by John Bull', lane.display_name)
+        eq_(mock_api, lane.novelist_api)
 
-        # Pagination works.
-        with self.request_context_with_library("/?size=1&order=title"):
-            response = self.manager.work_controller.recommendations(
-                self.identifier.type, self.identifier.identifier,
-                novelist_api=mock_api
-            )
+        facets = kwargs.pop('facets')
+        assert isinstance(facets, DatabaseBackedFacets)
+        eq_(Facets.ORDER_TITLE, facets.order)
+        eq_(AudiobooksEntryPoint, facets.entrypoint)
 
-        eq_(200, response.status_code)
-        feed = feedparser.parse(response.data)
-        eq_(1, len(feed['entries']))
-        [entry] = feed['entries']
-        eq_(another_work.title, entry['title'])
+        pagination = kwargs.pop('pagination')
+        eq_(30, pagination.offset)
+        eq_(2, pagination.size)
 
-        metadata.recommendations = [
-            self.work.license_pools[0].identifier,
-            another_work.license_pools[0].identifier,
-        ]
-        mock_api.setup(metadata)
+        annotator = kwargs.pop('annotator')
+        eq_(lane, annotator.lane)
 
-        with self.request_context_with_library("/?after=1&order=title"):
-            response = self.manager.work_controller.recommendations(
-                self.identifier.type, self.identifier.identifier,
-                novelist_api=mock_api
+        # Checking the URL is difficult because it requires a request
+        # context, _plus_ the DatabaseBackedFacets, Pagination and Lane
+        # created during the original request.
+        route, url_kwargs = lane.url_arguments
+        url_kwargs.update(dict(facets.items()))
+        url_kwargs.update(dict(pagination.items()))
+        with self.request_context_with_library(""):
+            expect_url = self.manager.work_controller.url_for(
+                route, library_short_name=library.short_name,
+                **url_kwargs
             )
-
-        eq_(200, response.status_code)
-        feed = feedparser.parse(response.data)
-        eq_(1, len(feed['entries']))
-        [entry] = feed['entries']
-        eq_(self.work.title, entry['title'])
+        eq_(kwargs.pop('url'), expect_url)
 
     def test_related_books(self):
```

Comment:
Now that all of the sublanes used by RelatedBooksLane have been converted over, I converted over its controller test as well.